File: API/models/pago.py
from typing import Optional

from pydantic import BaseModel


class PagoCreate(BaseModel):
    pedido_id: Optional[int] = None
    carrito_id: Optional[int] = None
    monto: float
    moneda: str = "MXN"
    estado: str = "pendiente"
    pasarela: Optional[str] = None
    referencia_externa: Optional[str] = None
    respuesta_proveedor: Optional[str] = None

File: Laravel/app/Models/Macuin/Pago.php
<?php

namespace App\Models\Macuin;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Pago extends Model
{
    protected $table = 'pagos';

    public $timestamps = false;

    protected $casts = [
        'monto' => 'decimal:2',
        'creado_en' => 'datetime',
    ];

    protected $fillable = [
        'pedido_id',
        'carrito_id',
        'monto',
        'moneda',
        'estado',
        'pasarela',
        'referencia_externa',
        'respuesta_proveedor',
    ];

    public function pedido(): BelongsTo
    {
        return $this->belongsTo(Pedido::class, 'pedido_id');
    }
}
